src/engine/Values.cpp

```diff
@@ -138,16 +138,28 @@ void Values::writeValues(IdTable* res, const Index& index,
                          const SparqlValues& values) {
   IdTableStatic<I> result = res->moveToStatic<I>();
   result.resize(values._values.size());
-  for (size_t rowIdx = 0; rowIdx < values._values.size(); rowIdx++) {
-    const vector<string> row = values._values[rowIdx];
+  size_t numActuallyWritten = 0;
+  size_t numSkipped = 0;
+  for (const auto& row : values._values) {
     for (size_t colIdx = 0; colIdx < result.cols(); colIdx++) {
       size_t id;
       if (!index.getVocab().getId(row[colIdx], &id)) {
-        AD_THROW(ad_semsearch::Exception::BAD_INPUT,
-                 "The word " + row[colIdx] + " is not part of the vocabulary.")
+        getWarnings().push_back("The word " + row[colIdx] +
+                                " is not part of the vocabulary.");
+        numSkipped++;
+        goto skipRow;
```

Comment:
Please add a comment so that it's clear that this is used simply to break out of the inner loop (and nothing worse) and that one finds the corresponding label a few lines further down